py-polars/polars/internals/dataframe/frame.py

```diff
@@ -4215,6 +4216,9 @@ def with_column(self, column: pli.Series | pli.Expr) -> DataFrame:
         Creating a new DataFrame using this method does not create a new copy of
         existing data.
 
+        .. deprecated:: 0.15.14
+            `with_column` will be removed in favour of the more generic `with_columns`.
```

Comment:
Deprecation warning uses "favor" (American spelling), let's be consistent in this deprecation message.

```suggestion
            `with_column` will be removed in favor of the more generic `with_columns`.
```

Reply:
Good point, fixed.